### PR TITLE
Remove duplicate math environments

### PR DESCRIPTION
**Pull request by gassmoeller**
The Sphinx `math` directive already includes the functionality of `align`, and while the duplicated math environments do not seem to be a problem for the html generation, it seems to crash the latexpdf build (if we want to build the sphinx documentation as pdf).

### CODE REVIEW

File: doc/sphinx/user/methods/basic-equations/adiabatic-heating.md [hunk 1]
```diff
@@ -3,9 +3,7 @@
 Other codes and texts sometimes make a simplification to the adiabatic heating term in the previous equation.
 If you assume the vertical component of the gradient of the *dynamic* pressure to be small compared to the gradient of the *total* pressure (in other words, the gradient is dominated by the gradient of the hydrostatic pressure), then $-\rho \mathbf g \approx \nabla \mathbf{p}$, and we have the following relation (the negative sign is due to $\mathbf g$ pointing downwards)
 ```{math}
-\begin{align}
  \alpha T \left( \mathbf u \cdot \nabla \mathbf p \right)   & \approx -\alpha \rho T \mathbf u \cdot \mathbf g.
```
Comment (cmills1095) [inline doc/sphinx/user/methods/basic-equations/adiabatic-heating.md:6]:
```suggestion
 \alpha T \left( \mathbf u \cdot \nabla \mathbf p \right)   \approx -\alpha \rho T \mathbf u \cdot \mathbf g.
```

**Comment by gassmoeller** [comment]
Do not merge this yet, I am experimenting with different ways to improve the equations.

**Review by bangerth — APPROVED** [review]
OK as is, if you wanted to take an incremental approach to improving things.

**Comment by gassmoeller** [comment]
Ok, this is ready from my side for now.

**Review by bangerth — APPROVED** [review]

**Pull request merged**